src/index.d.ts

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import * as monacoEditor from 'monaco-editor';
+
+declare class MonacoEditor extends React.Component<MonacoEditorProps> {
+  constructor(props: Partial<MonacoEditorProps>);
+
+  componentDidMount(): void;
+  componentDidUpdate(prevProps: Partial<MonacoEditorProps>): void;
+  componentWillUnmount(): void;
+
+  editorDidMount(editor: monacoEditor.editor.IEditor, monaco: typeof monacoEditor): void;
+  editorWillMount(monaco: any): void;
+
+  afterViewInit(): void;
+  initMonaco(): void;
+  destroyMonaco(): void;
+
+  render(): React.DOMElement<any, any>;
+
+  updateModel(value: string | null, original: string | null): void;
+}
+
+declare interface MonacoEditorProps {
+  width: string | number;
```

Comment:
As we can see in the `PropTypes` definition: https://github.com/superRaytin/react-monaco-editor/blob/bde80d6095e463e01383da6c7627a3d246237404/src/editor.js#L109-L120

these props should be optional:

```diff
- width: string | number;
+ width?: string | number;
```

Otherwise TypeScript will raise an error if you miss some props:

![image](https://user-images.githubusercontent.com/6076919/39793693-1f283f18-5379-11e8-9839-679ba4c96b68.png)

Reply:
Ahh, yeah. Can you send a PR?

Reply:
Actually I tried to use `Partial<T>` for that. We may add `| undefined` on every field.

Reply:
I think ? is preferred if there are no other uses of the same type but with different required fields.

Reply:
Yes. If we change that, we could remove the `Partial<T>` in some places.